### ISSUE TITLE
How to modify the loss function of the bounding box?

### ISSUE THREAD

**Issue by zxsitu** [ultralytics/yolov5#2480]
## ❔Question
Hi author, if I want to modify the loss function of the bounding box, do I modify the  "GIOU" in the [utils.py](https://github.com/ultralytics/yolov5/blob/111994956ede6964433e02bc3f64deb04b9916ac/utils/utils.py#L409-L413) file (as shown below) and replace all of them with "CIOU"? or "DIOU"? Thanks.

## Additional context
From line 409 to line 433

            # GIoU
            pxy = ps[:, :2].sigmoid() * 2. - 0.5
            pwh = (ps[:, 2:4].sigmoid() * 2) ** 2 * anchors[i]
            pbox = torch.cat((pxy, pwh), 1)  # predicted box
            giou = bbox_iou(pbox.t(), tbox[i], x1y1x2y2=False, GIoU=True)  # giou(prediction, target)
            lbox += (1.0 - giou).sum() if red == 'sum' else (1.0 - giou).mean()  # giou loss

            # Obj
            tobj[b, a, gj, gi] = (1.0 - model.gr) + model.gr * giou.detach().clamp(0).type(tobj.dtype)  # giou ratio

            # Class
            if model.nc > 1:  # cls loss (only if multiple classes)
                t = torch.full_like(ps[:, 5:], cn)  # targets
                t[range(nb), tcls[i]] = cp
                lcls += BCEcls(ps[:, 5:], t)  # BCE

            # Append targets to text file
            # with open('targets.txt', 'a') as file:
            #     [file.write('%11.5g ' * 4 % tuple(x) + '\n') for x in torch.cat((txy[i], twh[i]), 1)]

        lobj += BCEobj(pi[..., 4], tobj)  # obj loss

    lbox *= h['giou']
    lobj *= h['obj']
    lcls *= h['cls']

**Comment by glenn-jocher** [comment]
@ilem777 the code you pasted is out of date, so before you do anything you should git pull or git clone the repo again. To use CIoU or DIoU you simply set the relevant bbox_iou() function arguments to True:
https://github.com/ultralytics/yolov5/blob/ed2c74218d6d46605cc5fa68ce9bd6ece213abe4/utils/general.py#L315-L316

**Comment by zxsitu** [comment]
@glenn-jocher Thank you for your prompt reply, it was my fault.
I will try to revise it.

**Comment by jayyoung23** [comment]
@glenn-jocher I still have a question,in general.py bbox_iou is just a Defined function,but in loss.py L132:**iou = bbox_iou(pbox.T, tbox[i], x1y1x2y2=False, CIoU=True)** ,i find that **ciou=true**,so i think yolov5 uses **ciou** as default,but when i browse the web，i looked that someone says yolov5 uses **giou** as default,so now i am confused.hope you can reply me,thanks!

**Comment by glenn-jocher** [comment]
@jayyoung23 CIoU is the default box regression loss criteria, as you state.
https://github.com/ultralytics/yolov5/blob/005d7a8c54a39d89bf2b9dc03fba82a489cd0628/utils/loss.py#L128-L134
